core/src/main/scala/org/typelevel/otel4s/otel4s.scala

```diff
@@ -151,6 +156,15 @@ object Counter {
     new Counter[F, A] {
       def add(value: A, attribute: Attribute[_]*) = F.unit
     }
+
+  implicit final class CounterSyntax[F[_], A](
+      private val counter: Counter[F, A]
+  ) extends AnyVal {
+
+    def inc(attributes: Attribute[_]*)(implicit A: Numeric[A]): F[Unit] =
+      counter.add(A.one, attributes: _*)
```

Comment:
Yes, FWIW this seemed strange to me,.`Numeric` doesn't strike me as a catsy type-class :)

Reply:
That relates kind of to #4.  The Java backend just duplicates Long and Double all over the place.  I'm still not sure what we want to do.

Reply:
Since `A` is a type parameter of `Counter` itself, we could make these into abstract methods on it. To save duplicate implementations we could then have `LongCounter` and `DoubleCounter` that implementations can extend from, or something. But I can't see far enough ahead the implications of this.

Reply:
Sounds reasonable. 

`A: Numeric` implies the `Counter` can use any numeric type. 
It's possible indeed, but in reality, the constraint is broader than it should be. 
It all comes down to the otel Java, that currently offers only `LongCounter` and `DoubleCounter`.  

I made `inc` an abstract method. Could you take a look, please?

Reply:
Thanks! I like it 👍